### PR TITLE
Expanded tests and functionality for dask/kubernetes tests

### PR DESCRIPTION
**Pull request by kaipak**
- Made Dask/GCP tests more resilient and less messy in that it cleans up files when it's done and it'll shut down the cluster as well.
- Created function to wait until cluster is completely provisioned before running tests.
- Fixed various bugs with Xarray test
- Off loaded clean-up routines to storage API layer that were once in benchmark layer.
- Additional parameters for benchmarks.

### CODE REVIEW

File: IO_xarray.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        self.target = target_zarr.ZarrStore(backend=backend)
+        self.target.get_temp_filepath()
+        bmt.rand_xarray().to_zarr(self.target.storage_obj)
+
+    def time_synthetic_read(self, backend):
+        ds = xr.open_zarr(self.target.storage_obj).load()
+
+    def time_synthetic_mean(self, backend):
+        ds = xr.open_zarr(self.target.storage_obj).load()
+        ds.mean()
```
Comment (rabernat) [inline IO_xarray.py]: not sure about the size of the arrays getting created here, but calling `.load()` on the whole dataset will cause everything to load into memory. Calling `.mean()` after that doesn't have anything to do with IO performance, so I'm not really sure what this benchmark is measuring.

File: benchmarks/IO_dask.py [hunk 2]
```diff
@@ -42,76 +54,110 @@ def test_gcp():
...
+            self.da.store(self.target.storage_obj, lock=False)
+        else:
+            raise NotImplementedError("Not on GCP Pangeo environment...\
+                                      skipping")
+
+    def time_mean(self, backend, n_chunks, n_workers):
+        if self.is_gcp:
+            pretty_name = "10 GB - compute mean"
+            test_da = da.from_array(self.target.storage_obj, chunks=self.chunks)
+            test_da.mean().compute()
```
Comment (rabernat) [inline benchmarks/IO_dask.py]: 👍

File: benchmarks/IO_dask.py [hunk 3]
```diff
@@ -42,76 +54,110 @@ def test_gcp():
...
+            # Maybe combine above and below methods so init does all this?
+            # seems a little superfluous?
+            self.target.get_temp_filepath()
+            self.da.store(self.target.storage_obj, lock=False)
+        else:
+            raise NotImplementedError("Not on GCP Pangeo environment...\
+                                      skipping")
+
+    def time_mean(self, backend, n_chunks, n_workers):
+        if self.is_gcp:
```
Comment (rabernat) [inline benchmarks/IO_dask.py]: Maybe we could define a `@gcp` decorator, rather than these big `if self.is_gcp` blocks?

Comment (kaipak) [reply benchmarks/IO_dask.py]: Suggestions are noted. I'll put in the requested changes and amend this PR. I'll also suppress all commits into single one so the log doesn't get cluttered.

**Pull request closed without merging**